src/behavior_generation_lecture_python/graph_search/a_star.py

```diff
@@ -131,6 +131,12 @@ def a_star(self, start: str, end: str) -> bool:
         :param end: Name of the end node.
         :return: True if shortest path found, False otherwise.
         """
+        try:
+            self.nodes_dict["K"]
+            self.nodes_dict[end]
+        except KeyError as error:
+            print(f"Could not find node {error} in the graph, Make sure that the start and end nodes are in the graph.")
```

Comment:
I propose to fail fast: 
`raise ValueError(f"Start and end node must be in graph. Detailed error: {error}.")`

Reply:
Alternatively, since I misuse the asserts also for input checking, 
```python
assert start in self.nodes_dict, "Start node must be in graph"
assert end in self.nodes_dict, "End node must be in graph"
```

Reply:
Yeah true.  I will change it.